system/jlib/jlog.hpp

```diff
@@ -829,7 +829,54 @@ constexpr LogMsgCategory MCauditInfo(MSGAUD_audit, MSGCLS_information, AudMsgThr
 constexpr LogMsgCategory MCstats(MSGAUD_operator, MSGCLS_progress, ProgressMsgThreshold);
 constexpr LogMsgCategory MCoperatorInfo(MSGAUD_operator, MSGCLS_information, InfoMsgThreshold);
 
-inline LogMsgCategory MCexception(IException * e, LogMsgClass cls = MSGCLS_error) { return LogMsgCategory((e)->errorAudience(),cls); }
+/*
+ * Function to determine log level (detail) for exceptions, based on log message class
+ */
+inline LogMsgDetail mapClassToDefaultDetailLevel(LogMsgClass cls)
+{
+    switch (cls)
+    {
+    case MSGCLS_disaster:
+    case MSGCLS_all:
+        return FatalMsgThreshold;
+    case MSGCLS_error:
+        return ErrMsgThreshold;
+    case MSGCLS_warning:
+        return WarnMsgThreshold;
+    case MSGCLS_information:
+        return InfoMsgThreshold;
+    case MSGCLS_progress:
+        return ProgressMsgThreshold;
+    default:
+        return DefaultDetail;
+    }
+}
+
+/*
+ * Function to determine log level (detail) for excpetion based log messages,
```

Comment:
Spotted this spelling error and will fix once @ghalliday determines if the changes in L:244-252 are within scope or not